libs/framework/include/celix/Bundle.h

```diff
@@ -74,6 +80,43 @@ namespace celix {
         }
 #endif
 
+       /**
+        * @brief Return a use-able entry path for the provided relative path to a bundle persistent storage.
+        *
+        * For example if there is a resource entry in the bundle persistent storage at path 'resources/counters.txt` this call
+        * will return a relative path to entry in the bundle persistent storage.
+        * .cache/bundle5/storage/resources/counters.txt
+        *
+        * A provided path is always relative to the bundle persistent storage root and can start with a "/".
+        * A provided path NULL, "", "." or "/" indicates the root of this bundle cache store.
+        *
+        * The returned entry path should can be treated as read-write.
+        *
+        * @param path The relative path to a bundle persistent storage entry.
+        * @return The use-able entry path or an empty string if the entry is not found.
+        */
+#if __cplusplus >= 201703L //C++17 or higher
+        [[nodiscard]] std::string getDataFile(std::string_view path) const {
+            std::string result{};
+            char* entry = celix_bundle_getDataFile(cBnd.get(), path.data());
+            if (entry != nullptr) {
+                result = std::string{entry};
+                free(entry);
```

Comment:
Raw resource not managed by RAII is not exception-safe.

Reply:
Of course. Added a try/catch and rethrow mechanism, I think this should solve the issue.

Reply:
What about unique_ptr with customized deleter?

Reply:
Yeah thanks, that is indeed cleaner. Will look into this.

Reply:
I counted 3 instances where a returned `char*` needed to be "converted" to a std::string. For now I would like keep the solution local, but I agree that if we see this more often a more general solution should be investigated.